File: 2-second-year-2022-2023/Programming-languages-and-laboratory/Python/Exercises/36_compito_python_20200116/Cliente.py
from Visita import Visita


class Cliente:
    def __init__(self, cognome, nome):
        self._cognome = cognome
        self._nome = nome
        self._visite = []
        self._totale_speso = 0

    def add_spesa(self, importo_speso):
        self._totale_speso += importo_speso

    def __str__(self):
        return self._cognome + "\t" + self._nome + "\t" + str(self._totale_speso)

File: Programming-languages-and-laboratory/Java/Exercises/36_compito_java_2020_01_16/Cliente.java
import java.util.*;

public class Cliente {
	protected String cognome, nome;
	protected List<Visita> visite;

	public Cliente(String cognome, String nome) {
		this.cognome = cognome;
		this.nome = nome;
		visite = new LinkedList<Visita>();
	}

	public float spesaTotale() {
		float totale = 0;
		for (Visita v : visite) {
			totale = totale + v.importoSpeso;
		}
		return totale;
	}

	public String toString() {
		return cognome + "\t" + nome + "\t" + spesaTotale();
	}
}